exchanges/alert/alert.go

```diff
@@ -1,23 +1,50 @@
 package alert
 
 import (
+	"errors"
 	"sync"
 	"sync/atomic"
 )
 
-// pool is a silent shared pool between all notice instances for alerting
-// external routines waiting on a state change.
-var pool = sync.Pool{New: func() interface{} { return make(chan bool) }}
-
 const (
 	inactive = uint32(iota)
 	active
 	alerting
 
 	dataToActuatorDefaultBuffer = 1
-	preAllocCommsDefaultBuffer  = 5
+	PreAllocCommsDefaultBuffer  = 5
+)
+
+var (
+	// pool is a silent shared pool between all notice instances for alerting
+	// external routines waiting on a state change.
+	pool = sync.Pool{New: func() interface{} { return make(chan bool) }}
+
+	preAllocBufferSize = PreAllocCommsDefaultBuffer
+	mu                 sync.RWMutex
+
+	errInvalidBufferSize = errors.New("invalid buffer size cannot be equal or less than zero")
 )
 
+// SetPreAllocationCommsBuffer sets buffer size of the pre-allocated comms.
+func SetPreAllocationCommsBuffer(size int) error {
+	if size <= 0 {
+		return errInvalidBufferSize
```

Comment:
While you have wrapped the err at the engine level, better here
```suggestion
		return fmt.Errorf("%w received %v", errInvalidBufferSize size)
```